### DISCUSSION TITLE
use MultiSamlStrategy in nestjs

### DISCUSSION THREAD

**Discussion by ZigBalthazar** [bergie/passport-saml#910] [Q&A]
Hi
i am having trouble defining the MultiSamlStrategy in NestJS
```ts
@Injectable()
export class samlStrategy extends PassportStrategy(MultiSamlStrategy, 'saml') {
  constructor() {
    super({
        //
    });
  }

  async validate(profile: any): Promise<any> {
    // Implement your validation logic here
    return profile;
  }
}
```
When using MultiSamlStrategy in NestJS to define multiple IDP, passing the config in `super()` is not possible.
How can I use MultiSamlStrategy in NestJS?
Can you provide me with an example?

**Comment by srd90** [comment]
Dunno but person who presented this question was able to get it to work eventually
* https://github.com/node-saml/passport-saml/discussions/873

and from constructor point of view that person's example code looks a lot like yours.

For the record: aforementioned question's root cause was https://github.com/node-saml/passport-saml/discussions/873#discussioncomment-6613582
